### PR TITLE
PIMS-631: Table Component

### PR DESCRIPTION
**Pull request by GrahamS-Quartech**
## 🎯 Summary
[PIMS-631](https://apps.itsm.gov.bc.ca/jira/browse/PIMS-631)

* Installed MUI's x-data-grid and material design icons. If we want different icons or no icons let me know but this seemed like a safe choice.
* Added some basic styling and a custom menu component that can be opened from a button in the table rows.
* Added some example usage with some placeholder data, status chips, and actions menu under the `/dev` route.

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.

### CODE REVIEW

File: react-app/src/themes/appTheme.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { createTheme } from '@mui/material';
 import '@bcgov/bc-sans/css/BC_Sans.css';
+import type {} from '@mui/x-data-grid/themeAugmentation';
```
Comment (dbarkowsky) [inline react-app/src/themes/appTheme.ts:3]: I thought this was odd, but it's official documentation. 🤷

File: react-app/src/components/table/DataTable.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+          </MenuItem>
+        ))}
+      </Menu>
+    </>
+  );
+};
+
+export const CustomDataGrid = (props: DataGridProps) => {
+  return (
+    <DataGrid {...props} slots={{ noRowsOverlay: NoRowsOverlay, loadingOverlay: LinearProgress }} />
```
Comment (dbarkowsky) [inline react-app/src/components/table/DataTable.tsx:101]: Like the use of LinearProgress.

**Review by dbarkowsky — APPROVED** [review]
Looks good. Works as expected.
Glad we're going with MUI Data Grid.

File: react-app/src/themes/appTheme.ts (hunk 1 above)
Comment (GrahamS-Quartech) [reply react-app/src/themes/appTheme.ts:3]: Yeah, not really sure how this one works exactly, but that's what it says...

**Comment by dbarkowsky** [comment]
Tests were failing after recent merge. I think the issue is with the moduleNameMapper in jest.config.ts. It only appears to recognize the first match, so it matches the import for pictures matching `@/assets/images...` but then fails to activate the following map that would replace it with a mock. Not sure how to address this yet, so I've just gutted that test for now.

**Pull request merged**